core/query/src/main/java/org/eclipse/rdf4j/query/QueryResults.java

```diff
@@ -66,7 +68,7 @@
 	 */
 	public static Model asModel(CloseableIteration<? extends Statement, ? extends RDF4JException> iteration)
 			throws QueryEvaluationException {
-		Model model = new LinkedHashModel();
+		Model model = new SimpleUpgradeableModel(new LinkedHashModelFactory());
```

Comment:
While we're at it, perhaps we could make the `ModelFactory` injectable here as well, by introducing a method overload. E.g. something like this:

    public static Model asModel(CloseableIteration<? extends Statement, ? extends RDF4JException> iteration) {
          return asModel(iteration, new SimpleUpgradeableModelFactory());
     }
     public static Model asModel(CloseableIteration<? extends Statement, ? extends RDF4JException> iteration, ModelFactory modelFactory) {
                     Model model = modelFactory.createEmptyModel();
                     ....